<?xml version="1.0"?>

<launch>
	<!-- rostopic pub /maurice/left_wheel_velocity_controller/command std_msgs/Float64 "data: 1.0" -->

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find maurice_control)/config/maurice_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
	output="screen" ns="/maurice" args="joint_state_controller
					  right_wheel_velocity_controller
					  left_wheel_velocity_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
	respawn="false" output="screen">
    <remap from="/joint_states" to="/maurice/joint_states" />
  </node>

  <!-- launch diff drive cmd vel -->
  <node name="diffdrive_cmdvel" pkg="maurice_control" type="diffdrive_cmdvel" output="screen">
    <param name="WHEEL_BASE" value="0.49" />
    <param name="WHEEL_RADIUS" value="0.125" />
    <param name="CMD_VEL_TOPIC" value="/cmd_vel" />
    <param name="LEFT_WHEEL_TOPIC" value="/maurice/left_wheel_velocity_controller/command" />
    <param name="RIGHT_WHEEL_TOPIC" value="/maurice/right_wheel_velocity_controller/command" />
  </node>

</launch>
